Sources/BraintreeApplePay/BTApplePayClient.swift

```diff
@@ -115,11 +110,25 @@ import BraintreeCore
         ]
     }
 
-    func completionHandler(
-        onMainThreadWithPaymentRequest paymentRequest: PKPaymentRequest?,
-        error: Error?,
+    // MARK: - Analytics Completion Helpers
+
+    private func completionWithAnalytics(
+        for analyticsResult: BTAnalyticsResult,
+        nonce: BTApplePayCardNonce? = nil,
+        error: Error? = nil,
+        completion: @escaping (BTApplePayCardNonce?, Error?) -> Void
+    ) {
+        apiClient.sendAnalyticsEvent("apple-pay:tokenize:\(analyticsResult == .success ? "succeeded" : "failed")")
+        completion(nonce, error)
+    }
+
+    private func paymentRequestCompletionWithAnalytics(
```

Comment:
I also think we should sync with PPCP on our naming convention & pattern for returning these compeltions & analytic events, whatever we do land on.

Reply:
Yeah this one is a bit of an edge case because there are 2 public methods in apple pay returning different results (one a nonce + error and the other a paymentRequest + error). I think we should align them for completions that don't notify a delegate but _do_ think the notify language makes more sense for methods that do bubble up to a delegate as it makes it more clear what is actually happening (notifying a delegate vs just calling a completion).